docs/windows-design.md

```diff
@@ -0,0 +1,260 @@
+by kubelet on Windows.
+
+### OVS Management
+OVS is working as Windows Service. Two corresponding services are registered: ovsdb-server and ovs-vswitchd.
+
+## Traffic walk though
+
+### Intra Cluster Traffic
+
+The traffic of Pod to local Pod and Pod to remote Pod is the same as Antrea on Linux. 
+It is processed and forwarded by OVS, and controlled with OpenFlow entries.
+
+
+### Service Traffic
+
+Kube-proxy userspace mode is configured to provide the Service functions. A specific Network Adapter named as
+"HNS Internal NIC" is provided to kube-proxy to configure Service addresses. The OpenFlow entries for the Service
+traffic on Windows are the same as those on Linux.
+
+The traffic of Pod to Service is similar to Antrea on Linux. The packet from Pod to Service leaves OVS pipeline
+on the gateway interface first. kube-proxy captures the packet on the host and then chooses a valid endpoint. 
+The destination IP of the packet is changed to the endpoint, and enters the OVS pipeline again.
+
+A difference is the packet is using the client Pod's IP as source address when it enters the OVS pipeline at the
+second time on Linux, but it uses the gateway IP as source on Windows. This is because kube-proxy on Windows sets
+up a new connection on the gateway interface to the chosen backend Pod, and the packet is forwarded back to OVS
+from the new connection.
+
+### External Traffic
+
+The traffic of Pod to external address is SNATed in OVS using the Node internal IP first, and then it leaves
+the OVS pipeline on the gateway interface. As IP-Forwarding on the gateway interface is enabled, the packet
+is forwarded to the OVS bridge interface and enters the OVS for the second time. 
+When the packet enters the OVS at the second time, the dMAC is changed to an appropriate value using the host
+network stack. And OVS outputs the packet to the uplink interface directly.
+
+For the reply packet of Pod to external address, it enters the OVS from the uplink interface. It will enter
+the ct zone, and the destination IP address is recovered to the Pod IP within the ct context. Then it will be
+output to the Pod.
+
+<img src="/docs/assets/windows_external_traffic.svg" width="600" alt="Traffic to external">
+
+### Host Traffic
+
+With “Transparent” mode, Antrea agent should also processes the host traffic if necessary, including the packets
+sent from the host to external, and the ones sent from the external to the host. 
+
+The host traffic is similar to the Pod external traffic, for the destination IP is not in the cluster CIDR, but
+it is not sent from local Pod. OVS check the packet source firstly, if it is sent from the OVS bridge, and it is
+not sent to local Pod, it is output to the uplink interface directly. The reply packet is checked in the ct
+context first, then goes through the OVS pipeline if it has been set with ct_mark. The packet that is not marked
+in the ct context is output to the Windows host from OVS bridge interface. 
```

Comment:
The host traffic case is similar to the Pod-to-external traffic, as the destination IP is not in the cluster CIDR, but we can differentiate based on the source IP. If the packet is sent from the OVS bridge interface, and if it is not destined to a local Pod, it is output to the uplink interface directly. Reply traffic goes through conntrack first, then through the rest of OVS pipeline if the ct_mark is set. If the mark is not set, it is output to the Windows host on the OVS bridge interface.